src/_P003_Pulse.ino

```diff
@@ -268,10 +270,9 @@ boolean Plugin_003(byte function, struct EventStruct *event, String& string)
         P003_Step3OKcounter[event->TaskIndex]   = P003_pulseTotalCounter[event->TaskIndex];
         resetStatsErrorVars(event->TaskIndex);
       #endif // P003_PULSE_STATISTIC
-
-      P003_log = F("INIT : PulsePin: ");
-      P003_log += Settings.TaskDevicePin1[event->TaskIndex];
-      addLog(LOG_LEVEL_INFO, P003_log);
+      String log;
+      log = F("INIT : PulsePin: "); log += Settings.TaskDevicePin1[event->TaskIndex];
+      addLog(LOG_LEVEL_INFO, log);
```

Comment:
Same here and also other parts where you create logs.

N.B. When wrapping it in if (...) { ... } checks, the allocation of the String object is in a smaller scope.
If you are handling a switch case, you must place the part between `case BLA:` the next `case BLA2:` in `{}` if you construct new objects.

Reply:
OK, I hope I caught now all places.

I don't understand, what you are referring to with your N.B. remark. I assume, its a general remark.
I thing the logging is now included in {...}. New version comes soon.